test/hotspot/jtreg/testlibrary_tests/template_framework/examples/TestTutorial.java

```diff
@@ -0,0 +1,555 @@
+            """
+            package p.xyz;
+
+            public class InnerTest6 {
+                public static void main() {
+                    System.out.println("Welcome to main!");
+                    """,
+                    template1.fillWith(0),
+                    """
+                }
+            }
+            """
+        ));
+
+        // Render templateClass to String.
+        return templateClass.render();
+    }
+
+    // In the example below ("generateWithNames"), we see the use of Names to add
+    // variables and fields to code scopes, and then sample from those variables
+    // and fields. Every Name has a Name.Type, which we now define for int and long.
+    private record MyPrimitive(String name) implements Name.Type {
+        @Override
+        public boolean isSubtypeOf(Name.Type other) {
+            return other instanceof MyPrimitive(String n) && n == name();
+        }
+
+        @Override
+        public String toString() { return name(); }
+    }
+    private static final MyPrimitive myInt = new MyPrimitive("int");
+    private static final MyPrimitive myLong = new MyPrimitive("long");
+
+    // Example with names, i.e. addName, weighNames, and sampleName.
+    // These can be used to add variables and fields to code scopes, and then sample
+    // from the available variables and fields later.
+    public static String generateWithNames() {
+        var templateSample = Template.make("type", (Name.Type type) -> body(
+            let("name", sampleName(type, false).name()),
+            """
+            System.out.println("Sampling type #type: #name = " + #name);
+            """
+        ));
+
+        var templateStaticField = Template.make("type", (Name.Type type) -> body(
+            addName(new Name($("field"), type, true, 1)),
+            """
+            public static #type $field = 0;
+            """
+        ));
+
+        var templateLocalVariable = Template.make("type", (Name.Type type) -> body(
+            addName(new Name($("var"), type, true, 1)),
+            """
+            #type $var = 0;
+            """
+        ));
+
+        var templateStatus = Template.make(() -> body(
+            let("ints", weighNames(myInt, false)),
+            let("longs", weighNames(myLong, false)),
+            """
+            System.out.println("Status: #ints ints, #longs longs.");
+            """
+        ));
+
+        var templateMain = Template.make(() -> body(
+            """
+            System.out.println("Starting inside main...");
+            """,
+            templateStatus,
+            Hooks.METHOD_HOOK.insert(templateLocalVariable.fillWith(myInt)),
+            Hooks.METHOD_HOOK.insert(templateLocalVariable.fillWith(myLong)),
+            Hooks.CLASS_HOOK.insert(templateStaticField.fillWith(myInt)),
+            Hooks.CLASS_HOOK.insert(templateStaticField.fillWith(myLong)),
+            templateStatus,
+            // We should see a mix if fields and variables sampled.
```

Comment:
Just guessing from the context:

```suggestion
            // We should see a mix of fields and variables sampled.
```